### DISCUSSION TITLE
Is it possible to run the performance dump from the package?

### DISCUSSION THREAD

**Discussion by Deom23** [mattdodge/yahoofantasy#36] [Q&A]
I am probably just not seeing how it works properly, but is there a way to run the performances dump from within package? The issue I am running into is that I am trying to run 'yahoofantasy dump ....' command from a python os.system() command, which cannot accept user inputs, inserting which league you want to gather the performances for. 

Can I run this somehow from the package where I can give it which league I want the dump for? Does it always return a file, or can it return a dataframe in session?

**Comment by mattdodge** [comment]
Interesting question - there's not really a direct way to do it I'm afraid, most of those `dump` commands assume they are going to be writing to a file. I can think of two approaches that might work
1. Run the corresponding method from `yahoofantasy.cli.dump` and then read the file from the filesystem afterwards. A bit ugly, but it might work
2. Grab the code from the CLI method and run it separately. Most of them aren't too complicated.

It sounds like you're wondering about player performance dumps so [here's the method](https://github.com/mattdodge/yahoofantasy/blob/223a8d44a537c67abb77ec2d821eb4e7e3f165a5/yahoofantasy/cli/dump.py#L100-L116) where this happens as well as a code snippet that might help

```python
from yahoofantasy.cli.dump import _get_results
results = []
league = ctx.get_leagues('nfl', 2022)[0]  # consider replacing with your league object
for week in league.weeks():
    if week.week_num > league.current_week or week.matchups[0].status != 'postevent':
        continue
    for matchup in week.matchups:
        results.extend(_get_results(matchup.team1, week.week_num))
        results.extend(_get_results(matchup.team2, week.week_num))
```
